substratevm/src/com.oracle.svm.configure/src/com/oracle/svm/configure/config/SerializationConfiguration.java

```diff
@@ -85,7 +86,8 @@ public void registerWithTargetConstructorClass(ConfigurationCondition condition,
     }
 
     @Override
-    public void registerWithTargetConstructorClass(ConfigurationCondition condition, String className, String customTargetConstructorClassName) {
+    public void registerWithTargetConstructorClass(ConfigurationCondition condition, String rawClassName, String customTargetConstructorClassName) {
+        String className = rawClassName.contains("$$Lambda$") ? rawClassName.split(LambdaUtils.SPLIT_BY_LAMBDA)[0] + "$$Lambda$" : rawClassName;
```

Comment:
"$$Lambda$" is a good candidate for `LambdaUtils`

Reply:
Added to LambdaUtils as suggested.